lib/discourse.rb

```diff
@@ -113,16 +113,19 @@ def self.activate_plugins!
   end
 
   def self.recently_readonly?
-    return false unless @last_read_only
-    @last_read_only > 15.seconds.ago
+    recently_readonly = $redis.get(recently_readonly_mode_key)
```

Comment:
This was sneaky.. I kept getting randomly failing tests until I realized this variable is shared between multisite.